pkg/distributor/query.go

```diff
@@ -145,6 +145,7 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 			if err == io.EOF {
 				break
 			} else if err != nil {
+				ingesterQueryFailures.WithLabelValues(ing.Addr).Inc()
```

Comment:
Can we have this check at a higher level, like after `replication.Do` returns?

Reply:
The metric tracks the failures per ingester. If we do after `replication.Do` returns, we don't have the detail about which ingester failed. Moreover, this metric should be specular to `ingesterQueries` metric, which is tracked within the same function.